### PR TITLE
 Bug 1242905 - Convert clonejobs.js to ReactJS components

### PR DESCRIPTION
**Pull request by camd**
Ready for merge when it's been soaked on prototype and we feel confident in the new Auth0 branch in production.

Co-authored-by: Casey Williams <cwillia5@gmail.com>

### CODE REVIEW

File: ui/job-view/jobGroup.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    }
+    return { buttons, counts };
+  }
+
+  render() {
+    this.items = this.groupButtonsAndCounts();
+
+    return (
+      <span className="platform-group">
+            <span className="disabled job-group"
```
Comment (edmorley) [inline ui/job-view/jobGroup.jsx]: I think this is indented a bit too much? (https://github.com/airbnb/javascript/tree/master/react#alignment)

File: ui/job-view/repo.jsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      this.$rootScope.$emit(this.thEvents.jobClick, job);
+    }, 200);
+  }
+
+  render() {
+    return null;
+  }
+}
+
+JobGroup.propTypes = {
```
Comment (edmorley) [inline ui/job-view/repo.jsx]: Should this be `Repo.propTypes`?

File: ui/entry-index.js [hunk 3]
```diff
@@ -83,14 +86,14 @@ require('./js/controllers/filters.js');
 require('./js/controllers/jobs.js');
 require('./js/controllers/bugfiler.js');
 require('./js/controllers/tcjobactions.js');
+require('./plugins/pinboard.js');
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/entry-index.js]: I think they must have been changed by something I did to auto-fix for linting.  But setting them back doesn't create a complaint, so I'll set them back to reduce the change size.

File: ui/index.html [hunk 4]
```diff
@@ -40,77 +40,6 @@
...
-        <script type="'text/ng-template'" id="runnableJobBtnClone.html">
-            <button class="btn runnable-job-btn {{ key }}  {{ extraClasses }}"
-                  data-jmkey="{{ key }}"
-                  data-buildername="{{ buildername }}"
-                  data-ignore-job-clear-on-click
-                  title="{{ title }}">{{ value }}</button>
-        </script>
-
         <!-- Tooltip for job info-->
         <script type="'text/ng-template'" id="jobInfoTooltip.html">
```
[the remark this answers is not in the document]
Comment (camd) [reply ui/index.html]: good point, might as well remove it...

**Comment by camd** [comment]
Thank you for finding all those issues!  I believe I have fixed them all now.

File: ui/js/models/resultsets_store.js [hunk 5]
```diff
@@ -854,28 +854,6 @@ treeherder.factory('ThResultSetStore', [
...
-            var rs = repositories[repoName].rsMap[resultsetId].rs_obj;
-            if (rs && rs.revisions.length === 0) {
-                $log.debug("loadRevisions: check out to load revisions", rs, repoName);
-                // these revisions have never been loaded; do so now.
-                return ThResultSetModel.get(rs.revisions_uri)
-                    .then(({ data }) => {
-
-                        if (rs.revisions.length === 0) {
-                            Array.prototype.push.apply(rs.revisions, data);
-                            $timeout($rootScope.$emit(thEvents.revisionsLoaded, rs));
```
Comment (edmorley) [inline ui/js/models/resultsets_store.js]: `revisionsLoaded` in `thEvents` is unused after this change

File: ui/entry-index.js [hunk 6]
```diff
@@ -34,6 +34,10 @@ require('ngreact');
 require('jquery.scrollto');
 require('./vendor/resizer.js');
 
+// Treeherder React UI
+require('./job-view/Repo.jsx');
+require('./job-view/Push.jsx');
```
Comment (edmorley) [inline ui/entry-index.js]: `s/\.jsx//`

(The rest in this file we can sort out sometime later)

**Comment by edmorley** [comment]
@camd, the last two review comments above are present in the latest update btw.

**Comment by camd** [comment]
@edmorley : Oops thanks for keeping me honest.  :)  OK, fixed those.

**Pull request merged**